### DISCUSSION TITLE
Recurring tasks / reminders

### DISCUSSION THREAD

**Discussion by latvia234** [athensresearch/athens#1418] [Feature Requests]
I would like the option to insert a recurring reminder into a block via the / keybinding.

Maybe this option could insert something like this:
{{[[TODO]][[2018-12-21]][[2d]]}} Every 2 days
{{[[TODO]][[date]][[4w]]}} Every 4 weeks on this day
{{[[TODO]][[date]][[3w135]]}} Every 3 weeks on Mondays, Wednesdays, and Fridays
{{[[TODO]][[date]][[5m]]}} Every 5 months on the same day [1]
{{[[TODO]][[date]][[1y]]}} Every year on the same day [2]
{{[[TODO]][[date]][[~30d]]}} Repeat in 30 days after completing the item

By default, checking off a recurring date will generate the next occurrence as a new item, right underneath the original item.

This is how Dynalist flavored markdown looks for recurring tasks:
!(2018-12-21 | 2d) | Every 2 days
!(2018-12-21 | 4w) | Every 4 weeks on this day
!(2018-12-21 | 3w135) | Every 3 weeks on Mondays, Wednesdays, and Fridays
!(2018-12-21 | 5m) | Every 5 months on the same day [1]
!(2018-12-21 | 1y) | Every year on the same day [2]
!(2018-12-21 | ~30d) | Repeat in 30 days after completing the item

- 1: In this case, months that do no contain that date will be skipped.
- 2: "1y" is useful for birthdays and anniversaries.

Please take a look at https://help.dynalist.io/article/133-recurring-dates

**Comment by TillScout** [comment]
Agree, very nice. I would suggest to add additional shortcuts for first/last working day of the month.

**Comment by infogulch** [comment]
This might be integrated with [Basic Note Templates](https://github.com/athensresearch/athens/discussions/875) to instantiate a particular template on each occurrence. Some use cases:

* Build out a detailed daily routine and just repeat it without having to copy/paste or type it all out every day
* Maintain a note taking structure for a particular recurring meeting, each new meeting instance would already have the template ready
* Set up a template to record your workout routine, where every exercise is already written out and you only have to add time, reps, weight, etc

I think these are the problems that need to be solved before this story can work:

* Athens needs to support some kind of "uninstantiated note" that becomes instantiated upon interaction or a button or something, otherwise every recurring item would add an infinite number of notes (and consume an infinite amount of space) with notes out to the end of time -- not what is intended when a user adds something that recurs every 2 days and doesn't know when it will end.
  * I'm thinking it shows up sorta ghost-like or greyed out and it doesn't really "exist" until you click on it, then it becomes solid and is saved as a note on that day
  * This way you avoid the issue of deciding what happens when you edit the recurrence note. Any items that were instantiated are not changed (don't really want editing the past), any ones that haven't been instantiated yet are 'changed'.
* Some way to easily "cancel" the recurrence
  * I guess this could be done by linking back to the original note that creates the recurring items, and you can just edit it to not recurr anymore.

**Comment by jefftangx** [comment]
Curious if plain-text representation is the right UX for this. What do you think @latvia234 ?

**Reply by latvia234** [reply]
@tangjeff0 

This depends on the target audience that Athens wants to serve. I read the last blog post that you wrote and found this:
> [The goal for Athens is to serve everyone.](https://athensresearch.ghost.io/season-2/)

If everyone includes my mother and grandmother it could be a challenge for them to work with plain-text only.

So the question could be: How can recurring tasks be made simple and powerful at the same time?

Todoist made it really simple to just type, and it will trigger on some words like "every". However it could be annoying to have this inside Athens because most of the time you're not adding recurring dates. Maybe it is a solution to only trigger this when there is already a date inside the same block?

https://user-images.githubusercontent.com/86680397/130366411-73dd50b0-5bd7-4766-ba9d-68c128d27d7c.mov

**Reply by latvia234** [reply]
Just using [[Every 2 days]] instead of [[2d]]. Could this be an option?

{{[[TODO]][[date]][[Every 2 days]]}}
{{[[TODO]][[date]][[Every 4 weeks]]}}
{{[[TODO]][[date]][[Every 5 months]]}}
{{[[TODO]][[date]][[Every year]]}}
By default, checking off a recurring date will generate the next occurrence as a new item, right underneath the original item.

When clicking on [[Every XYZ]] it will only show a dialog with all the recurring options. All other pages won't be visible.

![Schermafbeelding 2021-08-22 om 21 28 01](https://user-images.githubusercontent.com/86680397/130367716-c098606a-21e1-457e-9ad1-eb8907cc153b.png)

I guess that these options are too long:
{{[[TODO]][[date]][[Every 3 weeks on Mondays, Wednesdays, and Fridays]]}} 
{{[[TODO]][[date]][[Repeat in 30 days after completing the item]]}}
